### PR TITLE
Add filter property to MyGroupsSidebarItem

### PR DESCRIPTION
**Pull request by kielosz**
Signed-off-by: kielosz <kielosz@gmail.com>

## Hey, I just made a Pull Request!

Added the possibility to provide an additional filter when using `MyGroupsSidebarItem`, so we can have a customized view, e.g. some companies may want to filter by `spec.type` to display only certain types of groups:

``` diff
// app/src/components/Root/Root.tsx
<SidebarPage>
    <Sidebar>
      //...
      <SidebarGroup label="Menu" icon={<MenuIcon />}>
        {/* Global nav, not org-specific */}
        //...
        <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
        <MyGroupsSidebarItem
          singularTitle="My Squad"
          pluralTitle="My Squads"
          icon={GroupIcon}
+         filter={{ 'spec.type': 'team' }}
        />
       //...
      </SidebarGroup>
    </ Sidebar>
</SidebarPage>
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/spicy-forks-return.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-org': patch
+---
+
+Added the ability to use an additional `filter` when fetching groups in `MyGroupsSidebarItem` component
```
Comment (awanlin) [inline .changeset/spicy-forks-return.md:5]: Would it be possible to include an example here? Would be handy!

**Review by awanlin — APPROVED** [review]
This is cool and a great addition thanks for this contribution!

**Review by jhaals — APPROVED** [review]
Nice work, thank you 🎉

**Comment by suuus** [comment]
Thanks for your contribution @kielosz 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**